Sources/Logging/Locks.swift

```diff
@@ -68,21 +72,28 @@ internal final class Lock {
     }
 
     deinit {
-        #if os(Windows)
+        #if canImport(WASILibc)
+        // WASILibc is single threaded, provides no locks
+        #elseif os(Windows)
         // SRWLOCK does not need to be free'd
         #else
         let err = pthread_mutex_destroy(self.mutex)
         precondition(err == 0, "\(#function) failed in pthread_mutex with error \(err)")
         #endif
+
+        #if !canImport(WASILibc)
         self.mutex.deallocate()
+        #endif
```

Comment:
should `self.mutex.deallocate()` be moved into the condition instead?

or alternatively change deallocate to handle WASI? I assume that is what is happening on Windows?

Reply:
Unless I'm mistaken, deallocate is from UnsafeMutablePointer, so I don't think that can be changed here. I put this in its own condition to avoid the duplication, but I can move it into the Windows and default cases above later today. Same for rwlock further down.

Reply:
oh I see. yes I think its best to be symmetrical and deallocate inside the same branches as allocating

Reply:
okay, I updated the PR accordingly